src/components/chatbox/utils.ts

```diff
@@ -17,10 +24,19 @@ export function useScrollBarVisible() {
 		}
 
 		function checkScrollbarVisibility() {
-			setIsScrollbarVisible(
-				element.scrollHeight > element.clientHeight ||
-					element.scrollWidth > element.clientWidth
-			)
+			// is the element scrollable?
+			const hasScrollbarY = element.scrollHeight > element.clientHeight
+			setIsScrollbarYVisible(hasScrollbarY)
+
+			// has scrolling reached the end?
+
+			const atEnd =
+				/* total elem height - current scroll position */
+				element.scrollHeight - element.scrollTop <=
+				/* visible elem height + 5px buffer */
+				element.clientHeight + 5
+
+			setReachedYEnd(atEnd)
```

Comment:
Addresses buggy behavior observed in
[✓ Global: Scroll tool- displaying at the end of scroll and in scenarios where there's no scroll required](https://app.asana.com/0/0/1205523765248042/f)